src/commonMain/kotlin/com/charleskorn/kaml/YamlNodeReader.kt

```diff
@@ -29,37 +29,43 @@ import it.krzeminski.snakeyaml.engine.kmp.events.SequenceStartEvent
 internal class YamlNodeReader(
     private val parser: YamlParser,
     private val extensionDefinitionPrefix: String? = null,
-    private val allowAnchorsAndAliases: Boolean = false,
+    private val maxAliasCount: Int = 0,
 ) {
     private val aliases = mutableMapOf<Anchor, YamlNode>()
+    private val nodeWeight = mutableMapOf<YamlNode, Int>()
```

Comment:
Puh, don't ask me such things more than 6 weeks after I wrote it. :-D
I have no idea whether I had a reason or not.
It might be an oversight, or a left-over of a previous implementation, I changed it quite a bit before I was satisfied.